### PR TITLE
EntityReference widget + iFrame display working

### PR DESCRIPTION
**Pull request by slashrsm**
- selecting entities
- propagating selection to entity reference widget form elements
- saving references into a field

### CODE REVIEW

File: src/Plugin/EntityBrowser/Display/IFrame.php [hunk 1]
```diff
@@ -60,26 +153,32 @@ public function selectionCompleted(array $entities) {
...
           'js' => [
-            0 => [
+            [
               'type' => 'setting',
               'data' => [
                 'entity_browser' => [
-                  'iframe' => array_map(function (EntityInterface $item) {return [$item->id(), $item->uuid(), $item->getEntityTypeId()];}, $this->entities),
+                  'iframe' => [
+                    'entities' => array_map(function (EntityInterface $item) {return [$item->id(), $item->uuid(), $item->getEntityTypeId()];}, $this->entities),
+                    'uuid' => $_GET['uuid'],
```
Comment (marcingy) [inline src/Plugin/EntityBrowser/Display/IFrame.php]: Shouldn't this be using $request->query opposed to direct access?

File: src/Entity/EntityBrowser.php [hunk 2]
```diff
@@ -426,20 +426,25 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
...
 
-    $requirements = array(
+    $requirements = [
       '_permission' => 'access ' . String::checkPlain($this->id()) . ' entity browser pages',
-    );
+    ];
+
+    $options = [
+      '_admin_route' => TRUE,
+    ];
```
Comment (marcingy) [inline src/Entity/EntityBrowser.php]: Not a major issue but we aren't altering the options array and there is no processing required as such so why not just avoid the assignment?

File: src/Plugin/EntityBrowser/Display/IFrame.php [hunk 3]
```diff
@@ -28,25 +35,111 @@
...
+   */
+  public function displayEntityBrowser() {
+    /** @var \Drupal\entity_browser\Events\RegisterJSCallbacks $event */
+    $event = $this->eventDispatcher->dispatch(Events::REGISTER_JS_CALLBACKS, new RegisterJSCallbacks($this->configuration['entity_browser_id']));
+    $uuid = $this->uuid->generate();
+    return [
+      '#theme_wrappers' => ['container'],
+      'link' => [
+        '#type' => 'html_tag',
+        '#tag' => 'a',
```
Comment (marcingy) [inline src/Plugin/EntityBrowser/Display/IFrame.php:121]: Is this the normal way of building a link now in d8 in a render array? Seems strange using html_tag that is all!

File: src/Plugin/Field/FieldWidget/EntityReference.php [hunk 4]
```diff
@@ -18,23 +25,153 @@
...
+      'entity_browser' => $this->entityManager->getStorage('entity_browser')->load($this->getSetting('entity_browser'))->getDisplay()->displayEntityBrowser(),
+      '#attached' => [
+        'js' => [
+          drupal_get_path('module', 'entity_browser') . '/js/entity_browser.entity_reference.js',
+        ],
+      ],
+      'current' => [
+        // TODO - create better display of current entities (possibly using a view)
+        // See: https://www.drupal.org/node/2366241
+        '#markup' => '<strong>Selected:</strong> <div class="current-markup">' . implode(' ', $ids) . '</div>',
```
Comment (marcingy) [inline src/Plugin/Field/FieldWidget/EntityReference.php]: Also does Selected need to be translated!

Comment (slashrsm) [inline src/Plugin/Field/FieldWidget/EntityReference.php]: This is temporary. We need something smarter here (see referenced issue). Added t() anyway though.

File: src/Plugin/EntityBrowser/Display/IFrame.php (hunk 3 above)
Comment (slashrsm) [inline src/Plugin/EntityBrowser/Display/IFrame.php:121]: Wanted src="#something" only and didn't succeed with it. Do you know how to achieve that in D8?

Comment (marcingy) [inline src/Plugin/EntityBrowser/Display/IFrame.php:121]: I don't and nothing leaps out after looking at functions available

**Pull request merged**